### ISSUE TITLE
other output formats for tables

### ISSUE THREAD

**Issue by TDJorgensen** [rempsyc/lavaanExtra#33]
For review: openjournals/joss-reviews#5701

I love the table formats!  I would like to have them accessible for my LaTeX documents, since I no longer use Word (for the same DRY and modularity principles your paper mentions).  The [`semTable`](https://cran.r-project.org/package=semTable) package provides only LaTeX and HTML output (or saved as a CSV file) for tables of lavaan output.  The `knitr::kable()` function does provide LaTeX or Markdown formats (the latter can also be rendered as Word), but I have to specify all the table cells myself (rather than your nice row/column headers).  

So this is a feature request, perhaps not doable in the time it takes for a review.   It looks like the [`flextable`](https://cran.r-project.org/package=flextable) package does not output LaTeX, but does output Markdown.  There might be an easy way to at least make that available.  RMarkdown could output a LaTeX file to capture it from there...

**Comment by rempsyc** [comment]
Thank you, I appreciate it!

In theory, flextable can export to PDF, e.g., with `output: pdf_document` in an `rmarkdown` document. I've never worked with LaTeX myself (still stuck with Word unfortunately for track-change and collaborating with coauthors who don't use R and don't program) so I don't know how that translates to your exact format. The tables are, however, not as pretty (but still relatively intact):

![pdf_output](https://github.com/rempsyc/lavaanExtra/assets/13123390/91555e83-d021-4d44-a4e8-2a49ceb1f627)

Furthermore, when trying `nice_fit()`, I am getting an error that some symbols are not supported in LaTeX:

```r
! LaTeX Error: Unicode character χ (U+03C7)
               not set up for use with LaTeX.
```

However, [one source](https://bookdown.org/yihui/rmarkdown-cookbook/latex-unicode.html) explains
> If you run into an error like this, it probably means that you are using the default LaTeX engine, pdflatex, to render the (intermediate .tex) document to PDF, which was unable to process certain Unicode characters in your document. If that is the case, you may switch to xelatex or lualatex, e.g.,
> ```
> output:
>   pdf_document:
>     latex_engine: xelatex
> ```

Doing so I was able to get the fit table too.

![fit_table](https://github.com/rempsyc/lavaanExtra/assets/13123390/115a2f7f-05e7-4f13-927a-ad81e18fc0f3)

Resulting PDF: [test_pdf.pdf](https://github.com/rempsyc/lavaanExtra/files/12835013/test_pdf.pdf)

I am happy to learn more about your LaTeX setup to see how we could make it compatible if it does not work already as in the `rmarkdown` to PDF example.

**Comment by TDJorgensen** [comment]
> still stuck with Word unfortunately for track-change and collaborating with coauthors who don't use R and don't program

I won't try to convert you :-) but [Overleaf](https://www.overleaf.com/) provides a great environment for collaborating on a paper in LaTeX, including editing in either the plain-text syntax editor or making edits to a rich-text editor (wysiwyg like Word, or maybe wysiwym like [LyX](https://www.lyx.org/)).  Accounts are free, but the "pro" account also has a track-changes feature (and GitHub integration).

> some symbols are not supported in LaTeX

Yes, math symbols are typically added to LaTeX in math mode:

```
... model was rejected, $\chi^2 (36) = 116.26$, $p < .001$, 
and approximate fit was mediocre, RMSEA = 0.90, 90\% CI [0.07, 0.10], ...
```

When I use `knitr::kable()`, I create a matrix/table with character strings (e.g., `"$chi^2$"`) to be interpreted as LaTeX.

But the xelatex compilation is also a solution to continue including unicode, as is exporting to PDF which is easily included (as an image rather than a text-based table) in a LaTeX document.

> I am happy to learn more about your LaTeX setup to see how we could make it compatible if it does not work already as in the `rmarkdown` to PDF example

Honestly, I think if the `flextable` package already supports this, all you should need to do at this point is mention that in the paper and vignettes when you mention exporting to *.docx.

**Comment by rempsyc** [comment]
> I won't try to convert you :-) but [Overleaf](https://www.overleaf.com/) provides a great environment for collaborating on a paper in LaTeX

Hum, it definitely is in my long-term goals to move to fully open-source software, so Overleaf does look like a great starting point for writing! Thank you for the suggestion. Actually I just created my account and started a test project. I love that it recompiles automatically when saving! It is in this way more similar to traditional "what-you-see-is-what-you-get" text processing software. Sadly it seems the free version is only limited to one collaborator and that my university does not have a license :(

Also, I added the following sentence in the paper, README, and vignette:

> It will also render to PDF in an `rmarkdown` document with `output: pdf_document`, 
but using `latex_engine: xelatex` is necessary when including unicode symbols
in tables like with the `nice_fit()` function.
